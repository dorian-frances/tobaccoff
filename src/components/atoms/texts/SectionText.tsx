import { Text } from 'react-native';
import React from 'react';
import { FontsEnum } from '../../../assets/fonts/fonts.enum';

export type SectionTextProps = {
  text: string;
  fontSize: number;
};

const SectionText = ({
  text = 'Section text example',
  fontSize = 22,
}: SectionTextProps) => {
  return (
    <Text style={{ fontFamily: FontsEnum.MEDIUM, fontSize: fontSize }}>
      {text}
    </Text>
  );
};

export default SectionText;
