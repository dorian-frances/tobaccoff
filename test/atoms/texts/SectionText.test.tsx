import React from 'react';
import { render } from '@testing-library/react-native';
import SectionText from '../../../src/components/atoms/texts/SectionText';

describe('<SectionText/>', () => {
  const text = 'fake-text';
  const fontSize = 20;
  it('should render correctly', () => {
    const component = render(<SectionText text={text} fontSize={fontSize} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
