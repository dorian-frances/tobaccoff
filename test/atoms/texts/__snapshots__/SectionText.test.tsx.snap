// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionText/> should render correctly 1`] = `
<Text
  style={
    {
      "color": "#000000",
      "fontFamily": "Montserrat-Medium",
      "fontSize": 20,
    }
  }
>
  fake-text
</Text>
`;
